src/main/scala/org/broadinstitute/hail/variant/VariantSampleMatrix.scala

```diff
@@ -275,6 +291,8 @@ object VSMSubgen {
 class VariantSampleMatrix[T](val metadata: VariantMetadata,
   val rdd: RDD[(Variant, (Annotation, Iterable[T]))])
```

Comment:
I think this has to be `rdd: OrderedRDD[Locus, Variant, (Annotation, Iterable[T]))]`.  Then your require/FIXME below can go away.